From ec254db8ddaf5a644010614d275edc703567ae7b Mon Sep 17 00:00:00 2001
From: Yuan Juntao <juntaox.yuan@intel.com>
Date: Fri, 28 Nov 2014 10:48:31 +0800
Subject: [PATCH 1/1] mmc: flush eMMC cache before clock slows down.

It would be better to flush eMMC's cache before slow down its
clock. Otherwise, a tuning for eMMC may fail.

Change-Id: I2a98ca0059398afb3327fd7ba4305029a8acde4b
Signed-off-by: Yuan Juntao <juntaox.yuan@intel.com>
---
 drivers/mmc/core/core.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/mmc/core/core.c b/drivers/mmc/core/core.c
index f9be42c..36f991b 100644
--- a/drivers/mmc/core/core.c
+++ b/drivers/mmc/core/core.c
@@ -2335,7 +2335,6 @@ static int mmc_do_hw_reset(struct mmc_host *host, int check)
 		return -EOPNOTSUPP;
 
 	mmc_host_clk_hold(host);
-	mmc_set_clock(host, host->f_init);
 
 	/*
 	 * before HW reset card, cache needs to be flushed. Otherwise
@@ -2344,6 +2343,8 @@ static int mmc_do_hw_reset(struct mmc_host *host, int check)
 	 */
 	mmc_cache_ctrl(host, 0);
 
+	mmc_set_clock(host, host->f_init);
+
 	host->ops->hw_reset(host);
 
 	/* If the reset has happened, then a status command will fail */
-- 
1.9.1

